<html>
<head>
<title>TestNG:  TemplateManagementTests</title>
<link href="../testng.css" rel="stylesheet" type="text/css" />
<link href="../my-testng.css" rel="stylesheet" type="text/css" />

<style type="text/css">
.log { display: none;} 
.stack-trace { display: none;} 
</style>
<script type="text/javascript">
<!--
function flip(e) {
  current = e.style.display;
  if (current == 'block') {
    e.style.display = 'none';
    return 0;
  }
  else {
    e.style.display = 'block';
    return 1;
  }
}

function toggleBox(szDivId, elem, msg1, msg2)
{
  var res = -1;  if (document.getElementById) {
    res = flip(document.getElementById(szDivId));
  }
  else if (document.all) {
    // this is the way old msie versions work
    res = flip(document.all[szDivId]);
  }
  if(elem) {
    if(res == 0) elem.innerHTML = msg1; else elem.innerHTML = msg2;
  }

}

function toggleAllBoxes() {
  if (document.getElementsByTagName) {
    d = document.getElementsByTagName('div');
    for (i = 0; i < d.length; i++) {
      if (d[i].className == 'log') {
        flip(d[i]);
      }
    }
  }
}

// -->
</script>

</head>
<body>
<h2 align='center'>TemplateManagementTests</h2><table border='1' align="center">
<tr>
<td>Tests passed/Failed/Skipped:</td><td>30/0/0</td>
</tr><tr>
<td>Started on:</td><td>Thu Nov 01 12:08:20 IST 2018</td>
</tr>
<tr><td>Total time:</td><td>1 seconds (1969 ms)</td>
</tr><tr>
<td>Included groups:</td><td></td>
</tr><tr>
<td>Excluded groups:</td><td></td>
</tr>
</table><p/>
<small><i>(Hover the method name to see the test class name)</i></small><p/>
<table width='100%' border='1' class='invocation-passed'>
<tr><td colspan='4' align='center'><b>PASSED TESTS</b></td></tr>
<tr><td><b>Test method</b></td>
<td width="30%"><b>Exception</b></td>
<td width="10%"><b>Time (seconds)</b></td>
<td><b>Instance</b></td>
</tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplate()'><b>testAddTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@634851c1</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplate()'><b>testAddTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@52c0568c</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplate()'><b>testAddTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@620e5544</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException()'><b>testAddTemplateServerException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@633301e2</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while adding the template: T1.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:58)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException(TemplateManagerDAOImplTest.java:206)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s.&apos;INSERT INTO IDN_TEMPLATE_MGT (TENANT_ID, NAME, DESCRIPTION, TEMPLATE_SCRIPT) VALUES (?,?,?,?  )
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:302)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:50)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:244)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:270)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1506030949", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1506030949'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while adding the template: T1.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:58)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException(TemplateManagerDAOImplTest.java:206)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s.&apos;INSERT INTO IDN_TEMPLATE_MGT (TENANT_ID, NAME, DESCRIPTION, TEMPLATE_SCRIPT) VALUES (?,?,?,?  )
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:302)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:50)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:244)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:270)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException()'><b>testAddTemplateServerException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@49eac84f</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while adding the template: T2.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:58)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException(TemplateManagerDAOImplTest.java:206)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s.&apos;INSERT INTO IDN_TEMPLATE_MGT (TENANT_ID, NAME, DESCRIPTION, TEMPLATE_SCRIPT) VALUES (?,?,?,?  )
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:302)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:50)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:244)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:270)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace247981085", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace247981085'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while adding the template: T2.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:58)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException(TemplateManagerDAOImplTest.java:206)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s.&apos;INSERT INTO IDN_TEMPLATE_MGT (TENANT_ID, NAME, DESCRIPTION, TEMPLATE_SCRIPT) VALUES (?,?,?,?  )
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:302)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:50)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:244)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:270)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException()'><b>testAddTemplateServerException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@370fd959</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while adding the template: T3.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:58)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException(TemplateManagerDAOImplTest.java:206)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s.&apos;INSERT INTO IDN_TEMPLATE_MGT (TENANT_ID, NAME, DESCRIPTION, TEMPLATE_SCRIPT) VALUES (?,?,?,?  )
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:302)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:50)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:244)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:270)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1020982680", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1020982680'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while adding the template: T3.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:58)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testAddTemplateServerException(TemplateManagerDAOImplTest.java:206)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s.&apos;INSERT INTO IDN_TEMPLATE_MGT (TENANT_ID, NAME, DESCRIPTION, TEMPLATE_SCRIPT) VALUES (?,?,?,?  )
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:302)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.addTemplate(TemplateManagerDAOImpl.java:50)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:244)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeInsert(JdbcTemplate.java:270)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplate()'><b>testDeleteTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@16d4d353</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplate()'><b>testDeleteTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@331069bf</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplate()'><b>testDeleteTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@5c0d967e</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException()'><b>testDeleteTemplateDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@43adbe93</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while deleting template from DB for tenant ID: -1234 and template name: T1.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:155)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException(TemplateManagerDAOImplTest.java:384)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;DELETE FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID =?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:149)
	... 29 more
Caused by: java.sql.SQLException: Connection conn59: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.prepareStatement(DelegatingConnection.java:279)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:313)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1721508464", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1721508464'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while deleting template from DB for tenant ID: -1234 and template name: T1.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:155)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException(TemplateManagerDAOImplTest.java:384)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;DELETE FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID =?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:149)
	... 29 more
Caused by: java.sql.SQLException: Connection conn59: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.prepareStatement(DelegatingConnection.java:279)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:313)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException()'><b>testDeleteTemplateDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@5dc7d016</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while deleting template from DB for tenant ID: 1 and template name: T2.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:155)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException(TemplateManagerDAOImplTest.java:384)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;DELETE FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID =?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:149)
	... 29 more
Caused by: java.sql.SQLException: Connection conn65: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.prepareStatement(DelegatingConnection.java:279)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:313)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace386321141", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace386321141'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while deleting template from DB for tenant ID: 1 and template name: T2.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:155)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException(TemplateManagerDAOImplTest.java:384)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;DELETE FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID =?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:149)
	... 29 more
Caused by: java.sql.SQLException: Connection conn65: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.prepareStatement(DelegatingConnection.java:279)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:313)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException()'><b>testDeleteTemplateDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@160a5e87</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while deleting template from DB for tenant ID: -1234 and template name: T3.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:155)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException(TemplateManagerDAOImplTest.java:384)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;DELETE FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID =?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:149)
	... 29 more
Caused by: java.sql.SQLException: Connection conn71: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.prepareStatement(DelegatingConnection.java:279)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:313)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1470942118", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1470942118'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while deleting template from DB for tenant ID: -1234 and template name: T3.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:155)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testDeleteTemplateDataAccessException(TemplateManagerDAOImplTest.java:384)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;DELETE FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID =?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.deleteTemplate(TemplateManagerDAOImpl.java:149)
	... 29 more
Caused by: java.sql.SQLException: Connection conn71: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.prepareStatement(DelegatingConnection.java:279)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:313)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByName()'><b>testGetTemplateByName</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@782a0e57, -1234</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByName()'><b>testGetTemplateByName</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@61c9e646, 1</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByName()'><b>testGetTemplateByName</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@4dccaef5, -1234</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException()'><b>testGetTemplateByNameDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@75ed8f59, -1234</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while retrieving template from DB for tenant ID: -1234 and template name: T1.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:80)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException(TemplateManagerDAOImplTest.java:280)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;SELECT TEMPLATE_ID,TENANT_ID,NAME,DESCRIPTION,TEMPLATE_SCRIPT FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID=?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:205)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:68)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:186)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1564699077", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1564699077'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while retrieving template from DB for tenant ID: -1234 and template name: T1.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:80)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException(TemplateManagerDAOImplTest.java:280)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;SELECT TEMPLATE_ID,TENANT_ID,NAME,DESCRIPTION,TEMPLATE_SCRIPT FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID=?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:205)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:68)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:186)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException()'><b>testGetTemplateByNameDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@425531c9, 1</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while retrieving template from DB for tenant ID: 1 and template name: T2.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:80)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException(TemplateManagerDAOImplTest.java:280)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;SELECT TEMPLATE_ID,TENANT_ID,NAME,DESCRIPTION,TEMPLATE_SCRIPT FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID=?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:205)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:68)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:186)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace417132985", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace417132985'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while retrieving template from DB for tenant ID: 1 and template name: T2.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:80)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException(TemplateManagerDAOImplTest.java:280)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;SELECT TEMPLATE_ID,TENANT_ID,NAME,DESCRIPTION,TEMPLATE_SCRIPT FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID=?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:205)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:68)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:186)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException()'><b>testGetTemplateByNameDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: org.wso2.carbon.identity.template.mgt.model.Template@2a8ccb99, -1234</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while retrieving template from DB for tenant ID: -1234 and template name: T3.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:80)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException(TemplateManagerDAOImplTest.java:280)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;SELECT TEMPLATE_ID,TENANT_ID,NAME,DESCRIPTION,TEMPLATE_SCRIPT FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID=?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:205)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:68)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:186)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1793650630", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1793650630'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while retrieving template from DB for tenant ID: -1234 and template name: T3.
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:80)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateByNameDataAccessException(TemplateManagerDAOImplTest.java:280)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;SELECT TEMPLATE_ID,TENANT_ID,NAME,DESCRIPTION,TEMPLATE_SCRIPT FROM IDN_TEMPLATE_MGT WHERE NAME=? AND TENANT_ID=?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:205)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getTemplateByName(TemplateManagerDAOImpl.java:68)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.fetchSingleRecord(JdbcTemplate.java:186)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateList()'><b>testGetTemplateList</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: 0, 0, -1234, 0</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateList()'><b>testGetTemplateList</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: 1, 1, -1234, 1</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateList()'><b>testGetTemplateList</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: 10, 0, -1234, 3</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException()'><b>testGetTemplateListDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: 0, 0, -1234, 0</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while listing template from DB for tenantID: -1234
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:124)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException(TemplateManagerDAOImplTest.java:342)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error while getting the database connection metadata.
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:368)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isDBTypeOf(JdbcUtils.java:97)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isH2MySqlOrPostgresDB(JdbcUtils.java:51)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:92)
	... 29 more
Caused by: java.sql.SQLException: Connection conn131: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.getMetaData(DelegatingConnection.java:343)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:245)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:364)
	... 32 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace370563826", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace370563826'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while listing template from DB for tenantID: -1234
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:124)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException(TemplateManagerDAOImplTest.java:342)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error while getting the database connection metadata.
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:368)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isDBTypeOf(JdbcUtils.java:97)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isH2MySqlOrPostgresDB(JdbcUtils.java:51)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:92)
	... 29 more
Caused by: java.sql.SQLException: Connection conn131: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.getMetaData(DelegatingConnection.java:343)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:245)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:364)
	... 32 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException()'><b>testGetTemplateListDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: 1, 1, -1234, 1</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while listing template from DB for tenantID: -1234
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:124)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException(TemplateManagerDAOImplTest.java:342)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error while getting the database connection metadata.
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:368)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isDBTypeOf(JdbcUtils.java:97)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isH2MySqlOrPostgresDB(JdbcUtils.java:51)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:92)
	... 29 more
Caused by: java.sql.SQLException: Connection conn137: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.getMetaData(DelegatingConnection.java:343)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:245)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:364)
	... 32 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace331784903", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace331784903'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while listing template from DB for tenantID: -1234
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:124)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException(TemplateManagerDAOImplTest.java:342)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error while getting the database connection metadata.
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:368)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isDBTypeOf(JdbcUtils.java:97)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isH2MySqlOrPostgresDB(JdbcUtils.java:51)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:92)
	... 29 more
Caused by: java.sql.SQLException: Connection conn137: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.getMetaData(DelegatingConnection.java:343)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:245)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:364)
	... 32 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException()'><b>testGetTemplateListDataAccessException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: 10, 0, -1234, 3</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while listing template from DB for tenantID: -1234
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:124)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException(TemplateManagerDAOImplTest.java:342)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error while getting the database connection metadata.
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:368)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isDBTypeOf(JdbcUtils.java:97)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isH2MySqlOrPostgresDB(JdbcUtils.java:51)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:92)
	... 29 more
Caused by: java.sql.SQLException: Connection conn143: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.getMetaData(DelegatingConnection.java:343)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:245)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:364)
	... 32 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace1137409621", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace1137409621'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while listing template from DB for tenantID: -1234
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:124)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testGetTemplateListDataAccessException(TemplateManagerDAOImplTest.java:342)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error while getting the database connection metadata.
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:368)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isDBTypeOf(JdbcUtils.java:97)
	at org.wso2.carbon.identity.template.mgt.util.JdbcUtils.isH2MySqlOrPostgresDB(JdbcUtils.java:51)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.getAllTemplates(TemplateManagerDAOImpl.java:92)
	... 29 more
Caused by: java.sql.SQLException: Connection conn143: url=jdbc:h2:mem:testTemplateMgtDB user=USERNAME is closed.
	at org.apache.commons.dbcp.DelegatingConnection.checkOpen(DelegatingConnection.java:398)
	at org.apache.commons.dbcp.DelegatingConnection.getMetaData(DelegatingConnection.java:343)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.getMetaData(PoolingDataSource.java:245)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.getDriverName(JdbcTemplate.java:364)
	... 32 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplate()'><b>testUpdateTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: T1, org.wso2.carbon.identity.template.mgt.model.Template@1fb8b4d8, org.wso2.carbon.identity.template.mgt.model.Template@342ad2a3</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplate()'><b>testUpdateTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: T2, org.wso2.carbon.identity.template.mgt.model.Template@40a772db, org.wso2.carbon.identity.template.mgt.model.Template@523e0ffb</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplate()'><b>testUpdateTemplate</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: T3, org.wso2.carbon.identity.template.mgt.model.Template@8fa7f10, org.wso2.carbon.identity.template.mgt.model.Template@6aa70b66</td>
<td></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException()'><b>testUpdateTemplateServerException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: T1, org.wso2.carbon.identity.template.mgt.model.Template@11ed8d43, org.wso2.carbon.identity.template.mgt.model.Template@338e0952</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while updating the template: T1 Updated.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:141)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException(TemplateManagerDAOImplTest.java:240)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;UPDATE IDN_TEMPLATE_MGT SET NAME= ?, DESCRIPTION= ?, TEMPLATE_SCRIPT= ? WHERE TENANT_ID= ? AND NAME = ?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:133)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace822267089", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace822267089'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while updating the template: T1 Updated.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:141)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException(TemplateManagerDAOImplTest.java:240)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;UPDATE IDN_TEMPLATE_MGT SET NAME= ?, DESCRIPTION= ?, TEMPLATE_SCRIPT= ? WHERE TENANT_ID= ? AND NAME = ?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:133)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException()'><b>testUpdateTemplateServerException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: T2, org.wso2.carbon.identity.template.mgt.model.Template@4047829, org.wso2.carbon.identity.template.mgt.model.Template@5392abe7</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while updating the template: T2 Updated.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:141)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException(TemplateManagerDAOImplTest.java:240)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;UPDATE IDN_TEMPLATE_MGT SET NAME= ?, DESCRIPTION= ?, TEMPLATE_SCRIPT= ? WHERE TENANT_ID= ? AND NAME = ?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:133)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace608080514", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace608080514'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while updating the template: T2 Updated.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:141)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException(TemplateManagerDAOImplTest.java:240)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;UPDATE IDN_TEMPLATE_MGT SET NAME= ?, DESCRIPTION= ?, TEMPLATE_SCRIPT= ? WHERE TENANT_ID= ? AND NAME = ?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:133)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
<tr>
<td title='org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException()'><b>testUpdateTemplateServerException</b><br>Test class: org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest<br>Parameters: T3, org.wso2.carbon.identity.template.mgt.model.Template@1f80815b, org.wso2.carbon.identity.template.mgt.model.Template@5cf8a5c9</td>
<td><div><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while updating the template: T3 Updated.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:141)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException(TemplateManagerDAOImplTest.java:240)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;UPDATE IDN_TEMPLATE_MGT SET NAME= ?, DESCRIPTION= ?, TEMPLATE_SCRIPT= ? WHERE TENANT_ID= ? AND NAME = ?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:133)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
... Removed 28 stack frames</pre></div><a href='#' onClick='toggleBox("stack-trace2054075579", this, "Click to show all stack frames", "Click to hide stack frames")'>Click to show all stack frames</a>
<div class='stack-trace' id='stack-trace2054075579'><pre>org.wso2.carbon.identity.template.mgt.exception.TemplateManagementServerException: Error occurred while updating the template: T3 Updated.
	at org.wso2.carbon.identity.template.mgt.util.TemplateMgtUtils.handleServerException(TemplateMgtUtils.java:48)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:141)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest.testUpdateTemplateServerException(TemplateManagerDAOImplTest.java:240)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
	at org.testng.TestRunner.privateRun(TestRunner.java:648)
	at org.testng.TestRunner.run(TestRunner.java:505)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
	at org.testng.TestNG.runSuites(TestNG.java:1049)
	at org.testng.TestNG.run(TestNG.java:1017)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:283)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:75)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:120)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
Caused by: org.wso2.carbon.database.utils.jdbc.exceptions.DataAccessException: Error in performing Database query: &apos;%s&apos;UPDATE IDN_TEMPLATE_MGT SET NAME= ?, DESCRIPTION= ?, TEMPLATE_SCRIPT= ? WHERE TENANT_ID= ? AND NAME = ?
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:230)
	at org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImpl.updateTemplate(TemplateManagerDAOImpl.java:133)
	... 29 more
Caused by: java.sql.SQLException: Connection is closed.
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.checkOpen(PoolingDataSource.java:185)
	at org.apache.commons.dbcp.PoolingDataSource$PoolGuardConnectionWrapper.prepareStatement(PoolingDataSource.java:312)
	at org.wso2.carbon.database.utils.jdbc.JdbcTemplate.executeUpdate(JdbcTemplate.java:219)
	... 30 more
</pre></div></td>
<td>0</td>
<td>org.wso2.carbon.identity.template.mgt.dao.impl.TemplateManagerDAOImplTest@de0c402</td></tr>
</table><p>
</body>
</html>